### PR TITLE
DTSPO-13619 Jenkins Disk IAC

### PR DESCRIPTION
**Pull request by reespozzi**
Adding terraform code for jenkins disk (and related resource group). Will need importing into state with no terraform plan changes.
Includes slight restructure of code and some maybe overly verbose tf config (will know after first apply, but stops replacement in the meantime)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by reespozzi** [comment]
PTLSBOX disk is already imported into the state file

File: components/jenkins/local.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+locals {
```
Comment (timja) [inline components/jenkins/local.tf:1]: group locals close to they are used to make them easier to find and reason about.
you only need to move them to a separate file if they are used in multiple places

File: components/jenkins/variables.tf [hunk 2]
```diff
@@ -13,6 +13,11 @@ variable "location" {
   default     = "UK South"
 }
 
+variable "source_resource_id" {
```
Comment (timja) [inline components/jenkins/variables.tf:16]:
```suggestion
variable "jenkins_disk_source_resource_id" {
```

File: components/jenkins/disk.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  resource_group_name        = local.disk_rg_name
+  storage_account_type       = "Premium_LRS"
+  create_option              = "Copy"
+  hyper_v_generation         = "V1"
+  disk_iops_read_write       = "5000"
+  disk_size_gb               = "1024"
+  on_demand_bursting_enabled = false
+  trusted_launch_enabled     = false
+  tier                       = "P30"
+  source_resource_id         = var.source_resource_id
```
Comment (timja) [inline components/jenkins/disk.tf:13]: add a comment about this maybe?

**Review by timja — APPROVED** [review]

**Pull request merged**